spec/safe_statements_spec.rb

```diff
@@ -3763,6 +3925,30 @@ def up
               expect(locks_for_table(table_name, connection: alternate_connection)).to be_empty
             end
 
+            it "does not allow re-entrancy when lock escalation detected" do
```

Comment:
I know there's nothing in our usage that would change this, but could we add a test that exiting the inner call does not release the lock?

Reply:
We discussed this offline. Pushed up some more tests in fb6284b